api/lib/src/lp.rs

```diff
@@ -193,21 +235,29 @@ pub trait LpApi: SignedExtrinsicApi {
 		id: OrderId,
 		option_tick_range: Option<Range<Tick>>,
 		size_change: IncreaseOrDecrease<RangeOrderSize>,
-	) -> Result<Vec<types::RangeOrder>> {
+		wait_for: WaitFor,
+	) -> Result<ApiWaitForResult<Vec<types::RangeOrder>>> {
 		// Submit the mint order
-		let (_tx_hash, events, ..) = self
-			.submit_signed_extrinsic(pallet_cf_pools::Call::update_range_order {
-				base_asset,
-				pair_asset,
-				id,
-				option_tick_range,
-				size_change,
-			})
-			.await
-			.until_in_block()
+		let xt_result = self
+			.submit_signed_extrinsic_wait_for(
+				pallet_cf_pools::Call::update_range_order {
+					base_asset,
+					pair_asset,
+					id,
+					option_tick_range,
+					size_change,
+				},
+				wait_for,
+			)
 			.await?;
 
-		Ok(collect_range_order_returns(events))
+		Ok(match xt_result {
```

Comment:
this map could be cut off into a function, one that Takes a WaitForResult and turns it into a ApiWaitForResult and takes a closure for the Details case?